File: 1518_WaterBottles/1518_WaterBottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        ans = 0
        empty = 0
        while numBottles > 0:
            ans += numBottles
            newBottles = math.floor((numBottles + empty)/ numExchange)
            empty = (numBottles + empty) % numExchange
            numBottles = newBottles
        return ans

File: 1518_WaterBottles/1518_WaterBottles.js
/**
 * @param {number} numBottles
 * @param {number} numExchange
 * @return {number}
 */
var numWaterBottles = function (numBottles, numExchange) {
  let empty = 0;
  let ans = 0;

  while (numBottles > 0) {
    ans += numBottles;
    let newBottles = Math.floor((numBottles + empty) / numExchange);
    empty = (numBottles + empty) % numExchange;
    numBottles = newBottles;
  }

  return ans;
};
